Tools/Scripts/libraries/webkitapipy/webkitapipy/allow_unittest.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (C) 2025 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import tempfile
+from pathlib import Path
+from unittest import TestCase
+
+from .allow import AllowList, AllowedSPI, PermanentlyAllowedReason
+
+Toml = b'''
+[key1."rdar://123456789"]
+symbol = "_TemporarilyAllowedSymbol"
+selector = "_initWithTemporarilyAllowedData:"
+class = "NSTemporarilyAllowed"
+
+[key2.not-web-essential]
+symbols = ["_Permanent1", "_Permanent2"]
+'''
+
+A1 = AllowedSPI(key='key1', bug='rdar://123456789',
+                symbols=['_TemporarilyAllowedSymbol'],
+                selectors=['_initWithTemporarilyAllowedData:'],
+                classes=['NSTemporarilyAllowed'])
+A2 = AllowedSPI(key='key2', bug=PermanentlyAllowedReason.NOT_WEB_ESSENTIAL,
+                symbols=['_Permanent1', '_Permanent2'],
+                selectors=[], classes=[])
+
+
+class TestAllowList(TestCase):
+    def setUp(self):
+        self.tempfile = tempfile.NamedTemporaryFile(prefix='TestAllowList-')
```

Comment:
We need to close this file in the `tearDown`, otherwise we'll get `ResourceWarning`.